runhouse/main.py

```diff
@@ -666,6 +351,182 @@ def cluster_list(
+        current_cluster.keep_warm(mins=mins)
+        # TODO: keep warm when we are in the cluster?
+    except ValueError as e:
+        if str(e) == f"Resource {cluster_name} not found":
+            console.print("Cluster is not saved in den.")
+            # TODO: do we want to check if the cluster is a live sky cluster, and inform the user how to keep_warm using sky?
+        else:
+            console.print(f"Failed to keep the cluster warm: {e}")
+    except Exception as e:
+        console.print(f"Failed to keep the cluster warm: {e}")
+
+
+@cluster_app.command("logs")
+def cluster_logs(
+    cluster_name: str = typer.Argument(
+        None,
+        help="Name of cluster to get the logs from. If not specified, we will get the logs of the local cluster.",
+    ),
+    tail: Optional[int] = typer.Option(
+        DEFAULT_SURFACED_LOG_LENGTH,
+        "-n",
+        "--tail",
+        help="Number of lines to show from the end of the logs",
+    ),
+):
+    """Load the logs of the Runhouse daemon running on a cluster.
+
+    Example:
+        .. code-block:: console:
+
+            $ runhouse cluster logs /sashab/rh-basic-cpu
+
+    """
+
+    cluster_or_local = rh.here
+    bring_logs_from_den = False
+
+    if cluster_or_local == "file" and not cluster_name:
+        # If running outside the cluster must specify a cluster name
+        console.print("Missing argument `cluster_name`.")
+        return
+
+    elif cluster_name:
+
+        try:
+            current_cluster = cluster(name=cluster_name)
+        except ValueError:
+            console.print("Failed to load status for cluster.")
+            return
+
+        if not current_cluster.is_up():
+            print_bring_cluster_up_msg(
+                cluster_name=cluster_name, msg_prefix="Can't get live cluster logs"
+            )
+            bring_logs_from_den = True
+
+        try:
+            if current_cluster._http_client:
+                current_cluster._http_client.check_server()
+        except requests.exceptions.ConnectionError:
+            print_bring_cluster_up_msg(
+                cluster_name=cluster_name, msg_prefix="Can't get live cluster logs"
+            )
+            bring_logs_from_den = True
+
+    elif not cluster_or_local:
+        # todo: make func as the print_bring_cluster_up_msg
+        console.print(
+            "Can't get live cluster logs: \N{smiling face with horns} Runhouse Daemon is not running... \N{No Entry} \N{Runner}. "
+            "Start it with `runhouse restart` or specify a remote "
+            "cluster to poll with `runhouse status <cluster_name>`."
+        )
+        bring_logs_from_den = True
+
+    else:
+        # we are inside the cluster
+        current_cluster = cluster_or_local  # cluster_or_local = rh.here
+
+    try:
+        if bring_logs_from_den:
+            console.print("Getting the last chunk of logs saved in Den.")
+            get_logs_den_resp = current_cluster._get_saved_logs_from_den()
```

Comment:
for consistency wdyt about having this take the `tail` value, so the requested number of lines (from den) are returned?

Reply:
we could do that, but that may require few pings to den, in case that length of the latest logs saved in den is less than the provided tail size. @jlewitt1

Reply:
ah right ok then yea lets just keep as is and return the contents of the latest file